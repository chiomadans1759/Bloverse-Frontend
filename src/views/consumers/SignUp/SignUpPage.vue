<template>
  <div class="sign-up-container">
    <div class="top">
      <div>
        <span class="top-left-container">
        <a href="/login">
          <i class="fal fa-long-arrow-left fa-2x"></i>
          <span class="back-text">Back</span>
        </a>
        <router-link to="/" class="router-link">
          <img class="logo" src="@/assets/Bloverse-white.svg">
          <img class="blue-logo" src="@/assets/Bloverse.svg">
        </router-link>
      </span>
      </div>
    </div>
    <div class="sign-up-content d-flex flex-column justify-content-center align-items-center">
      <div>
        <h2 class="sign-up-intro-one">Join Us</h2>
        <h4 class="sign-up-intro-two">Lorem ipsum dolor sit amet, vel
          <br /> accumsanliberse ex, ea nec
        </h4>
        <h5 class="sign-up-intro-three">Sign Up With</h5>

        <div class="social-login">
          <Button customClass="btn btn-light sign-up-social-button sign-up-facebook-btn">
            <img src="@/assets/facebook-logo-light.svg">
          </Button>
          <br>
          <Button customClass="btn btn-light sign-up-social-button">
            <img src="@/assets/google-text-logo.png" height="25" width="60">
          </Button>
        </div>

        <div class="grid-container">
          <div class="grid-item">
            <hr class="hr-text" />
          </div>
          <div class="middle-grid-item">
            <p><strong>Or</strong></p>
          </div>
          <div class="grid-item">
            <hr class="hr-text" />
          </div>
      </div>
    </div>

    <div class="sign-up-form-section">
      <text-input
        id="first-name"
        name="firstName"
        type="text"
        placeholder="First name"
        :value="firstName"
        :onChange="onChange"
      />
      <text-input
        id="last-name"
        name="lastName"
        type="text"
        placeholder="Last name"
        :value="lastName"
        :onChange="onChange"
      />
      <text-input
        id="email"
        name="email"
        type="email"
        placeholder="Email"
        :value="email"
        :onChange="onChange"
      />
      <text-input
        id="password"
        name="password"
        type="password"
        placeholder="Password"
        :value="password"
        :onChange="onChange"
      />
      <Button customClass="btn btn-primary primary-button">Sign Up</Button>

      <h4 class="sign-up-terms-and-privacy">
        By creating an account, you agree to our
        <router-link to="/terms-and-conditions" class="sign-up-link-text">Terms</router-link>
        and
        <router-link to="/privacy-policies" class="sign-up-link-text">Privacy</router-link>
      </h4>

      <Button customClass="btn btn-outline-secondary secondary-button">
        <h2 class="sign-up-new-bloverse-text">Have an account?
          <router-link to="/login" class="signup-login-text">Login</router-link>
        </h2>
      </Button>
    </div>

      <div class="signup-creator-text">
        <h2 class="sign-up-creator-text">Content creator? 
        <router-link to="/creators/apply" class="router-link sign-up-apply-now-text">
          Apply
        </router-link>
        </h2>
      </div>
    </div>
    <creator-card />
    <div class="bottom">
      <p><span class="bloverse">Bloverse</span> {{year}}. All Right reserved</p>
    </div>
  </div>
</template>

<script>
import Button from "../../../components/Button/Button.vue";
import TextInput from "../../../components/TextInput/TextInput.vue";
import CreatorCard from "../../../components/CreatorCard/CreatorCard.vue";

export default {
  data() {
    return {
      firstName: "",
      lastName: "",
      email: "",
      password: "",
      year: new Date().getFullYear()
    };
  },
  components: {
    Button,
    TextInput,
    CreatorCard,
  },
  methods: {
    onChange: function(event) {
      this[event.target.name] = event.target.value;
    }
  },
  mounted() {
    /* eslint-disable */
    hideTawkBot();
    window.onload = function() {
      hideTawkBot();
    };
  },
  destroyed() {
    showTawkBot();
  }
};
</script>

<style lang="scss">
@import "./SignUpPage.scss";
</style>
