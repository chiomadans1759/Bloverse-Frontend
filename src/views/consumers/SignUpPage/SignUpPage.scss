@import 'sass-rem';

.sign-up-container {
  padding-top: rem(20px);
  padding-bottom: rem(20px);
  background-color: #333333;
  height: 100vh;
  overflow: hidden;
  color: white;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  background-attachment: fixed;
  background-image:
        linear-gradient(rgba(37, 42, 48, 0.2), rgba(37, 42, 48, 0.2)),
        url('../../../assets/signin-image.jpg');
}

.sign-up-content {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  margin: 5rem auto;
}

.sign-up-intro-one {
  font-size: rem(24px);
  font-weight: bolder;
  margin: rem(40px) 0 rem(8px) 0;
}

.sign-up-intro-two {
  font-size: rem(16px);
  margin-bottom: rem(24px);
}

.sign-up-intro-three {
  font-size: rem(16px);
  font-weight: bolder;
  margin-bottom: rem(8px);
}

.sign-up-social-button {
  width: rem(281px);
  height: rem(38px);
}

.sign-up-social-button:hover {
  background-color: #ffffff !important;
}

.sign-up-facebook-btn {
  margin-bottom: rem(-10px);
}

.sign-up-form-section {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

.sign-up-terms-and-privacy {
  font-size: rem(16px);
  margin: rem(8px) 0 rem(24px) 0;
  width: rem(281px);
}

.sign-up-link-text {
  color: #096DD9;
}

.signup-creator-text {
  width: rem(281px);
  text-align: left;
}

.sign-up-creator-text {
  font-size: rem(14px);
  text-align: left;
}

.sign-up-apply-now-text {
  font-size: rem(14px);
  color: #096DD9 !important;
}

.sign-up-new-bloverse-text {
  font-size: 14px;
  text-align: left;
  color: #ffffff;
}

// These styles apply to other pages 

.grid-container {
  display: grid;
  grid-template-columns: 0.5fr 0.5fr 1.9fr;
}

.grid-item {
  opacity: .5;
}

.middle-grid-item {
  text-align: center;
  font-size: rem(16px);
  margin: rem(24px) 0 rem(24px) 0;
}

.hr-text {
  line-height: 0em;
  position: relative;
  outline: 0;
  border: 0;
  text-align: left;
  height: rem(2em);
  opacity: .8;
  width: 100%;
  margin: rem(24px) 0 rem(24px) 0;
}

.hr-text:before {
  content: '';
  background: #ad938a;
  position: absolute;
  top: 50%;
  width: 100%;
  height: rem(1px);
}

.new-bloverse-text > .signup-login-text {
  color: #096DD9 !important;
}

.text-input {
  background-color: rgba(0, 0, 0, 0.45) !important;
  border-color: rgba(255, 255, 255, 0.15) !important;
  color: white !important;
}

.back-text {
  font-size: rem(16px);
}

.top-left-container {
  margin-left: 5%;
}

.top-left-container > a {
  color: #ffffff;
}

.top-left-container > a > svg {
  vertical-align: middle;
}

.top-left-container > a > span {
  padding-left: rem(10px);
}

.logo {
  height: rem(40px);
  padding-left: rem(20px);
}
