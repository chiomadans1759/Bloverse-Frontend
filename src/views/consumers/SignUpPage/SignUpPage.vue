<template>
  <div class="main-container">
    <div class="top">
      <div>
        <span class="top-left-container">
        <a href="/login">
          <i class="fal fa-long-arrow-left fa-2x"></i>
          <span>Back</span>
        </a>
        <router-link to="/" class="router-link">
          <img class="logo" src="@/assets/Logo - White.svg">
        </router-link>
      </span>
      </div>
    </div>
    <div class="content">
      <div class="mainSection">
        <h2 class="welcome-text"> Join Us </h2>
        <br />
        <h3>Lorem ipsum dolor sit amet, vel
          <br /> accumsanliberse ex, ea nec
        </h3>
        <br />
        <h3 class="sign-up-text">Sign Up With</h3>
        <br />

        <div class="social-login">
          <Button>
            <img src="@/assets/facebook-logo-light.svg">
          </Button>
          <br>
          <Button>
            <img src="@/assets/google-text-logo.png" height="25" width="60">
          </Button>
        </div>
        <br />
        <div class="grid-container">
        <div class="grid-item">
          <hr class="hr-text" />
        </div>
        <div class="middle-grid-item"><p>Or</p></div>
        <div class="grid-item">
          <hr class="hr-text" />
        </div>
      </div>
      <br />
      </div>
      <br />

      <div class="form-section">
        <text-input
          id="first-name"
          name="firstName"
          type="text"
          placeholder="First name"
          :value="firstName"
          :onChange="onChange"
          custom-class="text-input"
        />
        <br>
        <br>
        <text-input
          id="last-name"
          name="lastName"
          type="text"
          placeholder="Last name"
          :value="lastName"
          :onChange="onChange"
          custom-class="text-input"
        />
        <br>
        <br>
        <text-input
          id="email"
          name="email"
          type="email"
          placeholder="Email"
          :value="email"
          :onChange="onChange"
          custom-class="text-input"
        />
        <br>
        <br>
        <text-input
          id="password"
          name="password"
          type="password"
          placeholder="Password"
          :value="password"
          :onChange="onChange"
          custom-class="text-input"
        />
        <br>
        <br>
        <Button customClass="primary-button">Sign Up</Button>
        <br>
        <h4 class="terms-and-privacy">
          By creating an account, you agree to
          <br>our
          <router-link to="/terms-and-conditions" class="terms-text">Terms</router-link>
          and
          <router-link to="/privacy-policies" class="privacy-text">Privacy</router-link>
        </h4>
        <br>
        <Button customClass="secondary-button">
          <h2 class="new-bloverse-text">Have an account?
            <router-link to="/login" class="signup-text">Login</router-link>
          </h2>
        </Button>
      </div>
      <br/><br/>
      <div class="signup-creator-text">
        <h2 class="creator-text">Content creator? 
        <router-link to="/creators/apply" class="router-link apply-now-text">
          Apply
        </router-link>
        </h2>
      </div>
    </div>
  </div>
</template>

<script>
import Button from "../../../components/button/button.vue";
import TextInput from "../../../components/TextInput/TextInput.vue";

export default {
  data() {
    return {
      firstName: "",
      lastName: "",
      email: "",
      password: ""
    };
  },
  components: {
    Button,
    TextInput
  },
  methods: {
    onChange: function(event) {
      this[event.target.name] = event.target.value;
    }
  }
};
</script>

<style lang="scss">
@import "./SignUpPage.scss";
</style>
