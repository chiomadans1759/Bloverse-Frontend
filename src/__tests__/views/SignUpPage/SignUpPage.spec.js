import { mount, RouterLinkStub } from '@vue/test-utils'
import SignUpPage from '@/views/consumers/SignUpPage/SignUpPage.vue';

describe('Signup Page', () => {
  it('should render correctly', () => {

    const wrapper = mount(SignUpPage, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.find(RouterLinkStub).props().to).toBe('/')
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should update the page state when input is changed', () => {
    const wrapper = mount(SignUpPage, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    const firstNameInput = wrapper.find('#first-name');
    firstNameInput.element.value = 'firstname';
    firstNameInput.trigger('change');

    const lastNameInput = wrapper.find('#last-name');
    lastNameInput.element.value = 'lastname';
    lastNameInput.trigger('change');

    const emailInput = wrapper.find('#email');
    emailInput.element.value = 'test@email.com';
    emailInput.trigger('change');

    const passwordInput = wrapper.find('#password');
    passwordInput.element.value = 'test12345';
    passwordInput.trigger('change');

    expect(wrapper.vm.firstName).toBe('firstname');
    expect(wrapper.vm.lastName).toBe('lastname');
    expect(wrapper.vm.email).toBe('test@email.com');
    expect(wrapper.vm.password).toBe('test12345');
  });
});
