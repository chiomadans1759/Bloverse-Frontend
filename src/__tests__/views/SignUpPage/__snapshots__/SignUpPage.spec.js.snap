// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Page should render correctly 1`] = `
<div
  class="signup-container"
>
  <div
    class="top"
  >
    <span
      class="top-left-container"
    >
      <a
        href="/login"
      >
        <i
          class="fal fa-long-arrow-left fa-2x"
        />
         
        <span>
          Back
        </span>
      </a>
       
      <a
        class="router-link"
      >
        <img
          class="logo"
          src="@/assets/Logo - White.svg"
        />
      </a>
    </span>
  </div>
   
  <div
    class="content"
  >
    <div
      class="signup-intro-text"
    >
      <h2
        class="join-us-text"
      >
        Join us
      </h2>
       
      <br />
       
      <h3>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit
      </h3>
       
      <br />
       
      <h3
        class="sign-up-text"
      >
        Sign Up With
      </h3>
    </div>
     
    <br />
     
    <div
      class="social-login"
    >
      <div>
        <button
          class="defaultButton"
        >
          <img
            src="@/assets/facebook-logo-light.svg"
          />
        </button>
      </div>
       
      <br />
       
      <div>
        <button
          class="defaultButton"
        >
          <img
            height="25"
            src="@/assets/google-text-logo.png"
            width="60"
          />
        </button>
      </div>
    </div>
     
    <br />
     
    <div
      class="grid-container"
    >
      <div
        class="grid-item"
      >
        <hr
          class="hr-text"
        />
      </div>
       
      <div
        class="middle-grid-item"
      >
        <p>
          Or
        </p>
      </div>
       
      <div
        class="grid-item"
      >
        <hr
          class="hr-text"
        />
      </div>
    </div>
     
    <br />
     
    <div
      class="basic-signup"
    >
      <input
        class="defaultClass text-input"
        id="first-name"
        name="firstName"
        placeholder="First name"
        type="text"
      />
       
      <br />
       
      <br />
       
      <input
        class="defaultClass text-input"
        id="last-name"
        name="lastName"
        placeholder="Last name"
        type="text"
      />
       
      <br />
       
      <br />
       
      <input
        class="defaultClass text-input"
        id="email"
        name="email"
        placeholder="Email"
        type="email"
      />
       
      <br />
       
      <br />
       
      <input
        class="defaultClass text-input"
        id="password"
        name="password"
        placeholder="Password"
        type="password"
      />
       
      <br />
       
      <br />
       
      <div>
        <button
          class="defaultButton login-button"
        >
          Sign Up
        </button>
      </div>
       
      <br />
       
      <h4
        class="terms-and-privacy"
      >
        
        By creating an account, you agree to
        
        <br />
        our
        
        <span
          class="terms-text"
        >
          Terms
        </span>
        
        and
        
        <span
          class="privacy-text"
        >
          Privacy
        </span>
      </h4>
       
      <br />
       
      <div>
        <button
          class="defaultButton signup-button"
        >
          <h2
            class="new-bloverse-text"
          >
            Have an account?
          
            <a
              class="signup-text"
            >
              Login
            </a>
          </h2>
        </button>
      </div>
    </div>
     
    <br />
     
    <br />
     
    <h2
      class="creator-text"
    >
      Content creator?
      
      <a
        class="router-link apply-now-text"
      >
        Apply
      </a>
    </h2>
  </div>
</div>
`;
